platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/AnnotationManager.java

```diff
@@ -194,18 +194,40 @@ private Marker prepareMarker(BaseMarkerOptions markerOptions) {
     }
 
     MarkerView addMarker(@NonNull BaseMarkerViewOptions markerOptions, @NonNull MapboxMap mapboxMap) {
+        nativeMapView.addOnMapChangedListener(new MapView.OnMapChangedListener() {
+            @Override
+            public void onMapChanged(@MapView.MapChange int change) {
+                if (change == MapView.DID_FINISH_RENDERING_FRAME_FULLY_RENDERED) {
```

Comment:
We probably should batch this so that we do only one invalidate per render instead of one per marker added